src/key.rs

```diff
@@ -26,11 +27,36 @@ use std::str::FromStr;
 ///
 /// To parse a key use `FromStr` trait implementation: `"string".parse::<Key>()`.
 #[derive(Debug, Eq, PartialEq, PartialOrd, Ord, Hash, Clone)]
-pub struct Key {
+pub struct SimpleKey {
```

Comment:
How about `SingleKey`? Also the doccomments need to be updated.

Reply:
I was thinking of going a different direction and having `Key` and `KeyPath` or something.

One thing benefit of "SimpleKey" is that this is the term the grammar uses